### DISCUSSION TITLE
v8: I think `pageCount` shouldn't be part of the state

### DISCUSSION THREAD

**Discussion by Gusis93** [TanStack/table#3999] [Ideas] [answered]
When working with React In v7 `pageCount` was outside of the state and in my opinion this worked better for controlled tables.

The `pageCount` normally depends on an API response, not any kind of user interaction, now that its part of the state we need to have 2 kind of duplicated values for the same purpose. You can see it in the controlled pagination example where you need to:

- Have a controlled state JUST for `pageSize` and` pageIndex`
- Have a memoized value that includes the previously created `pageSize` and `pageIndex` + now the `pageCount` that depends on the API response and have it observe when anything in the state changes, so I modify the state when I paginate, the state is modified but then also the memoized value changes as it depends on the state but the information is the same

There is a need now to touch the table state in order to modify the `pageCount`. For example if I'm using an API that depends on the state values it makes abstracting the state into a custom hook hard because I need to create a state that only contains whats needed for the API then duplicate this state into a memoised value, await the response and include the `pageCount` in the state then render the table with that state. 

If it wouldn't be part of the state I would abstract the whole table state without being concerned about the `pageCount` at all because once the table is going to be rendered with data I know that data contains the `pageCount` I need so I just give it to the `useTableInstance` hook as an option

TLDR: the `pageCount` should be more related to the data than to the state

**Comment by tannerlinsley** [comment, marked as answer]
This is now implemented in the latest beta 🎉

**Reply by Gusis93** [reply]
Very nice!
